plugin/pkg/scheduler/scheduler.go

```diff
@@ -44,6 +50,33 @@ type Scheduler struct {
 	config *Config
 }
 
+// These are the functions which need to be provided in order to build a Scheduler configuration.
+// An implementation of this can be seen in factory.go.
+type Configurator interface {
```

Comment:
-1 for IConfiguration (scheduler.Configurator was one of my suggestions, which I'm not super proud of, but I still think it's better than IConfiguration).